### ISSUE TITLE
Focal loss without weight

### ISSUE THREAD

**Issue by huixiancheng** [dvlab-research/SphereFormer#38]
Although weight could be None, look like it's necessary for caculate focal loss.

https://github.com/dvlab-research/SphereFormer/blob/0ff1e0ae84a4280fbecc9cfc3ac3aec41d24eeba/train.py#L177-L178
~~~
def focal_loss(output, target, class_weight, ignore_label, gamma, need_softmax=True, eps=1e-8):
    mask = (target != ignore_label)
    output_valid = output[mask]
    if need_softmax:
        output_valid = F.softmax(output_valid, -1)
    target_valid = target[mask]
    p_t = output_valid[torch.arange(output_valid.shape[0], device=target_valid.device), target_valid] #[N, ]
    class_weight_per_sample = class_weight[target_valid]
    focal_weight_per_sample = (1.0 - p_t) ** gamma
    loss = -(class_weight_per_sample * focal_weight_per_sample * torch.log(p_t + eps)).sum() / (class_weight_per_sample.sum() + eps)
    return loss
~~~
Can it be simply removed directly like this?
~~~
def focal_loss(output, target, ignore_label, gamma, need_softmax=True, eps=1e-8):
    mask = (target != ignore_label)
    output_valid = output[mask]
    if need_softmax:
        output_valid = F.softmax(output_valid, -1)
    target_valid = target[mask]
    p_t = output_valid[torch.arange(output_valid.shape[0], device=target_valid.device), target_valid] #[N, ]
    focal_weight_per_sample = (1.0 - p_t) ** gamma
    loss = -(focal_weight_per_sample * torch.log(p_t + eps)).sum() / eps
    return loss
~~~
